projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-site-connection-notice.tsx

```diff
@@ -0,0 +1,140 @@
+
+const useSiteConnectionNotice = ( redBubbleAlerts: RedBubbleAlerts ) => {
+	const { setNotice, resetNotice } = useContext( NoticeContext );
+	const { apiRoot, apiNonce } = getMyJetpackWindowRestState();
+	const { isRegistered, isUserConnected, hasConnectedOwner } = useMyJetpackConnection();
+	const { siteIsRegistering, handleRegisterSite } = useConnection( {
+		skipUserConnection: true,
+		apiRoot,
+		apiNonce,
+	} );
+	const products = useAllProducts();
+	const navToConnection = useMyJetpackNavigate( MyJetpackRoutes.Connection );
+
+	useEffect( () => {
+		if ( ! Object.keys( redBubbleAlerts ).includes( 'missing-site-connection' ) ) {
+			return;
+		}
+
+		const productSlugsThatRequireUserConnection =
+			getProductSlugsThatRequireUserConnection( products );
+		const requiresUserConnection =
+			! hasConnectedOwner && ! isUserConnected && productSlugsThatRequireUserConnection.length > 0;
+
+		if ( ! requiresUserConnection && isRegistered ) {
+			return;
+		}
+
+		const onActionButtonClick = () => {
+			if ( requiresUserConnection ) {
+				navToConnection();
+			}
+
+			handleRegisterSite().then( () => {
+				resetNotice();
+				setNotice( {
+					message: __( 'Your site has been successfully connected.', 'jetpack-my-jetpack' ),
+					options: {
+						level: 'success',
+						actions: [],
+						priority: NOTICE_PRIORITY_HIGH,
+						hideCloseButton: false,
+						onClose: resetNotice,
+					},
+				} );
+			} );
+		};
+
+		const oneProductMessage = sprintf(
+			/* translators: placeholder is product name. */
+			__(
+				'Jetpack %s needs a user connection to WordPress.com to be able to work.',
+				'jetpack-my-jetpack'
+			),
+			productSlugsThatRequireUserConnection[ 0 ]
+		);
+
+		const userConnectionContent = {
+			message:
+				productSlugsThatRequireUserConnection.length === 1
+					? oneProductMessage
+					: __(
+							'Some products need a user connection to WordPress.com to be able to work.',
+							'jetpack-my-jetpack'
+					  ),
+			buttonLabel: __( 'Connect your user account', 'jetpack-my-jetpack' ),
+			title: __( 'Missing user connection', 'jetpack-my-jetpack' ),
+		};
+
+		const siteConnectionContent = {
+			message: __(
+				'Some products need a connection to WordPress.com to be able to work.',
+				'jetpack-my-jetpack'
+			),
+			buttonLabel: __( 'Connect your site', 'jetpack-my-jetpack' ),
+			title: __( 'Missing site connection', 'jetpack-my-jetpack' ),
+		};
+
+		const noticeOptions = {
+			level: 'info',
+			actions: [
+				{
+					label: requiresUserConnection
+						? userConnectionContent.buttonLabel
+						: siteConnectionContent.buttonLabel,
+					isLoading: siteIsRegistering,
+					loadingText: __( 'Conecting…', 'jetpack-my-jetpack' ),
+					onClick: onActionButtonClick,
+					noDefaultClasses: true,
+				},
+			],
+			// If this notice gets into a loading state, we want to show it above the rest
+			priority: NOTICE_PRIORITY_HIGH + ( siteIsRegistering ? 1 : 0 ),
```

Comment:
I wonder if we should add another priority something like `NOTICE_PRIORITY_URGENT`. I am not sure that's better 🤔  However if we do stuff like this in other areas it might be hard to keep track of notice priority if the constant numbers are not all in once place. What do you think?

Reply:
In this case, this is what I had in mind
```suggestion
			priority: siteIsRegistering ? NOTICE_PRIORITY_URGENT : NOTICE_PRIORITY_HIGH,
```

Reply:
I thought about including a new higher-priority const, but it didn't feel right. Reasoning:
- We'll eventually have new priority constants, that's not a problem. But updating the state of the button seems more like an update to the current state rather than a completely new notice.
- If we were just to track/log the priority at some point and check if it's different from the ones we have, we'd still know which variation it'd be (because of the first digit, like in status codes)